helevents/models.py

```diff
@@ -1,6 +1,128 @@
+from functools import reduce
+
+from django.conf import settings
+from django_orghierarchy.models import Organization
 from helusers.models import AbstractUser
 
-from events.permissions import UserModelPermissionMixin
+from events.models import PublicationStatus
+
+
+class UserModelPermissionMixin:
+    """Permission mixin for user models
+
+    A mixin class that provides permission check methods
+    for user models.
+    """
+
+    @property
+    def is_external(self):
+        """Check if the user is an external user"""
+        return (
+            not self.organization_memberships.exists()
+            and not self.admin_organizations.exists()
+        )
+
+    def is_admin_of(self, publisher):
+        """Check if current user is an admin user of the publisher organization"""
+        raise NotImplementedError()
+
+    def is_regular_user_of(self, publisher):
+        """Check if current user is a regular user of the publisher organization"""
+        raise NotImplementedError()
+
+    @property
+    def admin_organizations(self):
+        """
+        Get a queryset of organizations that the user is an admin of.
+
+        Is replaced by django_orghierarchy.models.Organization's
+        admin_organizations relation unless implemented in a subclass.
+        """
+        raise NotImplementedError()
+
+    @property
+    def organization_memberships(self):
+        """
+        Get a queryset of organizations that the user is a member of.
+
+        Is replaced by django_orghierarchy.models.Organization's
+        organization_memberships relation unless implemented
+        in a subclass.
+        """
+        raise NotImplementedError()
+
+    def can_create_event(self, publisher, publication_status):
+        """Check if current user can create an event with the given publisher and publication_status"""
+        if self.is_admin_of(publisher):
+            return True
+        # Non-admins can only create drafts.
+        return publication_status == PublicationStatus.DRAFT
```

Comment:
If I'm just looking at this method, it now looks like it should be possible for anyone to create a draft event for any publisher. I wonder if it would make sense to do almost the same check as in the `can_edit_event` though here we would just not check `created_by`. I guess you could refactor `can_edit_event` and `can_create_event` to use a shared function or something in that case.

I might also add a way (setting) to disable this external event create/edit feature in case we need to deploy the backend withouth this feature enabled or in the case that later on we need to quickly disable it.

```suggestion
    def can_create_event(self, publisher, publication_status):
        """Check if current user can create an event with the given publisher and publication_status"""
        if self.is_admin_of(publisher):
            return True

        # Non-admins can only create drafts.
        if publication_status == PublicationStatus.DRAFT:
            if (
                settings.ENABLE_EXTERNAL_USER_EVENT_CREATE and
                ( publisher is None or publisher.id == settings.USER_DEFAULT_ORGANIZATION_ID)
            ):
                # External users can only create drafts for the default organization.
                return self.is_external and created_by == self
            else:
                # Regular users can edit drafts from organizations they are members of.
                return self.is_regular_user_of(publisher)

        return False
```

Reply:
Yup, you're right; had a brainfart here. Made it use `can_edit_event` with `created_by` set to `self`. Also added an `ENABLE_EXTERNAL_USER_EVENTS` flag.